techniques/applications/packageManagement/1.3/packageManagement.st

```diff
@@ -81,33 +61,89 @@ bundle agent package_management_&RudderUniqueID& {
+
+      # Version
+      "version_description[${index_pkg}]" string     => "in latest available version",
+                                          ifvarclass => "version_latest_${index_pkg}.(state_present_${index_pkg}|state_upgrade_only_${index_pkg})";
+      "version_description[${index_pkg}]" string     => "in any version",
+                                          ifvarclass => "!version_latest_${index_pkg}.!version_specified_${index_pkg}";
+      "version_description[${index_pkg}]" string     => "in version ${version[${index_pkg}]}",
+                                          ifvarclass => "version_specified_${index_pkg}";
+      # Non compatibility check is done by the "package_state_options" this is just to inform the user
+      "version_description[${index_pkg}]" string     => "in latest available version is not supported and",
+                                          ifvarclass => "version_latest_${index_pkg}.!(state_present_${index_pkg}|state_upgrade_only_${index_pkg})";
+
+
   classes:
+    any::
+      "state_upgrade_only_${index_pkg}"     expression => strcmp("${state[${index_pkg}]}", "upgrade_only");
+      "state_present_${index_pkg}"          expression => strcmp("${state[${index_pkg}]}", "present");
+
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+    pass1::
       # This class is different from the one in ncf (where a version number and latest are considered "specified")
       "version_specified_${index_pkg}"      expression => strcmp("${version[${index_pkg}]}", "specific");
       "architecture_specified_${index_pkg}" expression => strcmp("${architecture[${index_pkg}]}", "specific");
       "posthook_specified_${index_pkg}"            not => strcmp("${posthook[${index_pkg}]}", "");
       "manager_specified_${index_pkg}"             not => strcmp("${manager[${index_pkg}]}", "default");
-      "state_present_${index_pkg}"          expression => strcmp("${state[${index_pkg}]}", "present");
       "version_latest_${index_pkg}"         expression => strcmp("${version[${index_pkg}]}", "latest");
       "allow_unstrusted_${index_pkg}"       expression => strcmp("${manager_allow_untrusted[${index_pkg}]}", "true");
       "manager_apt_${index_pkg}"            expression => strcmp("${manager[${index_pkg}]}", "apt");
       "apt_allow_untrusted_${index_pkg}"    expression => "allow_unstrusted_${index_pkg}.((debian.!manager_specified_${index_pkg})|manager_apt_${index_pkg})";
-
-      "pass3" expression => "pass2";
-      "pass2" expression => "pass1";
-      "pass1" expression => "any";
+      "already_installed_${index_pkg}"      expression => "${inner_already_installed_class_prefix_${index_pkg}}_kept";
 
   methods:
+    pass1::
+      # Force to audit mode to verify if packages are already installed (whatever their version is)
+      "remove_dry_run_mode"            usebundle => push_dry_run_mode("true");
```

Comment:
shouldn't it be "force dry_run_mode" ? even if it's only a comment, it would help